File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        mp = set(nums)
        res = 1
        for num in mp:
            if num - 1 not in mp:
                target = num
                seq = 0
                while target in mp:
                    target += 1
                    seq += 1
                if seq > res:
                    res = seq
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> bucket = new HashMap<Integer, Integer>();
		for (int i = 0; i < nums.length; i++) {
    		bucket.put(nums[i], i);
		}
		for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (bucket.containsKey(complement) && bucket.get(complement) != i) {
                return new int[]{i, bucket.get(complement)};
            }
        }

        return new int[0];
    }
}
